het_control/models/het_control_mlp_empirical.py

```diff
@@ -166,6 +174,9 @@ def _forward(
             or distance.isnan().any()  # It is the first iteration
             or self.n_agents == 1
         ):
+            distance = self.estimate_snd(input)
+            if update_estimate:
+                self.estimated_snd[:] = distance.detach()
```

Comment:
Could you expalin this a bit?

If those conditions are met, we can avoid computing $\widehat{\mathrm{SND}}$

Reply:
I did this to be able to log the estimated_snd during training when the  desired snd is `-1`.  Right now it logs Nan's. It was just to be able to see the evolution of snd while training as well. Can be remove if necessary.

Reply:
I see, but you can still see it under eval/snd no?